lib/symboldatabase.h

```diff
@@ -1355,12 +1355,12 @@ class CPPCHECKLIB SymbolDatabase {
     const Type* findType(const Token* startTok, const Scope* startScope, bool lookOutside = false) const;
     Type* findType(const Token* startTok, Scope* startScope, bool lookOutside = false)
     {
-        return const_cast<Type*>(this->findType(startTok, const_cast<const Scope*>(startScope), lookOutside));
+        return const_cast<Type*>(this->findType(startTok, static_cast<const Scope*>(startScope), lookOutside));
```

Comment:
I feel it's a code smell both with a const_cast and a static_cast here. With a static_cast you tell the compiler to not warn about any invalid conversions at all.. I want to have warnings if there is some invalid conversion.

Reply:
```cpp
extern void f(char* p);
void f(char* p)
{
    (void)const_cast<const char*>(p);
    (void)static_cast<const char*>(p);
    (void)const_cast<const int*>(p);
    (void)static_cast<const int*>(p);
}
```

```
<source>:6:11: error: const_cast from 'char *' to 'const int *' is not allowed
    6 |     (void)const_cast<const int*>(p);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
<source>:7:11: error: static_cast from 'char *' to 'const int *' is not allowed
    7 |     (void)static_cast<const int*>(p);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://godbolt.org/z/xbcEh63df

Reply:
startScope is a `const Scope*` as far as I see ? so why is the cast needed

Reply:
> startScope is a `const Scope*` as far as I see ? so why is the cast needed

There two different function signatures. It is hard to spot unless you look closely.